cores/epoxy/main.cpp

```diff
@@ -130,14 +135,20 @@ int unixhostduino_main(int argc, char** argv) {
   }
 }
 
+#if defined(__MINGW32__)
+// Weak references to main() do not work properly under MING32
+// Define EPOXY_DUINO_NO_MAIN if the calling code provides its own main().
+#    ifndef EPOXY_DUINO_NO_MAIN
```

Comment:
This is an interesting way to nest preprocessor macros, using spaces between the `#` and `ifndef`. I've not seen it before, but looking around, some people seem to use it (like the gcc/g++ team?). I'm not a fan of it because this breaks my ability to grep for `#ifndef`, `#define`, etc. I would much prefer the following:

```C++
#if defined(__MINGW32__)
  ...
  #ifndef EPOXY_DUINO_NO_MAIN
    ...
  #endif
#else
  ...
#endif
```

Anyway, I'm sending you some changes which makes most of this moot.

Reply:
BTW, in one of my commits, I renamed `EPOXY_DUINO_NO_MAIN` to `EPOXY_DUINO_USER_DEFINED_MAIN` because it seemed more self-descriptive of the purpose of the flag. And I renamed `unixhostduino_main()` to `epoxyduino_main()` and made it `static`, removing it from `Arduino.h`. It's an implementation detail so its symbol does not need to be exported, and making it static should make it easier for the compiler/linker to completely remove it if it is never used.